docs/design/kuadrant-api-v2.md

```diff
@@ -0,0 +1,178 @@
+
+## Problem Statement
+
+Currently, the [latest](https://github.com/Kuadrant/kuadrant-controller/releases/tag/v0.2.0) release of
+kuadrant API is based on [apiproducts.v1beta1.networking.kuadrant.io](https://github.com/Kuadrant/kuadrant-controller/blob/v0.2.0/doc/architecture.md#api-product-crd)
+and [apis.v1beta1.networking.kuadrant.io](https://github.com/Kuadrant/kuadrant-controller/blob/v0.2.0/doc/architecture.md#api-crd)
+custom kubernetes resource APIs. Together with these CRDs, there is a bunch of kuadrant service
+annotations and labels available in order to provide the kuadrant
+[service discovery](https://github.com/Kuadrant/kuadrant-controller/blob/v0.2.0/doc/service-discovery.md)
+feature. All together, kuadrant provides routing, basic and global rate limiting and, finally,
+global authN (not authZ) features to protect customer owned APIs. While, good enough for a wide set
+of use cases, there are several limitations:
+
+* Rate limit is enforced to all services and endpoints defined in a APIProduct. It cannot be specified route/path wise.
+* Only high level rate limit directives are defined (`global`, `perRemoteIP` and `authenticated`). There is no option to specify with a fine-grained rate limit API.
+* Authentication is enforced to all services and endpoints defined in a APIProduct. It cannot be specified route/path wise.
+* Authentication is limited to `API key` and `OpenID Connect`.
+* Authorization is not available.
+
+## High Level Goals
+
+* Expose simple API, API management oriented, to protect customer owned API's.
+* Simple API can be extended with a fine-grained API to leverage full featured Authorino API and Limitador API.
+* Path/Operation based protection (i.e. authentication and rate limit)
+
+## Proposed Solution
+
+The kuadrant API will be oriented to meet API management requirements. In that context, the kuadrant
+API will be designed around the OpenAPI standard.
+Specifically, [OpenAPI 3.X specification](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md).
+Kuadrant will define the *APIProduct* object class (a CRD in terms of kubernetes) to wrap a set of
+API management configurations on top of a list of customer owened API definitions.
+
+### API Product
+
+**Custom Resource Definition GVK**
+
+Currently, as of kuadrant v0.2.0, APIProduct [Group Version Kind](https://book.kubebuilder.io/cronjob-tutorial/gvks.html)
+is: [apiproducts.v1beta1.networking.kuadrant.io](https://github.com/Kuadrant/kuadrant-controller/blob/v0.2.0/doc/architecture.md#api-product-crd)
+
+Taking into account kuadrant is not providing networking level API, but **API Management** level API,
+the proposed GVK is: **apiproducts.v1alpha1.apim.kuadrant.io**.
+
+**Domains**
+
+APIProduct resource will be configured with a list of destination domains or hostnames to which
+traffic is being sent.
+
+It is the responsability of cluster operators and API owners to forward *îngress* traffic for those
```

Comment:
+1 we shouldn't be creating the route IMO the application developer should define that